cmd/internal/cli/remote.go

```diff
@@ -110,47 +124,69 @@ var RemoteAddCmd = &cobra.Command{
 	Args:   cobra.ExactArgs(2),
 	PreRun: setGlobalRemoteConfig,
 	Run: func(cmd *cobra.Command, args []string) {
-		if err := singularity.RemoteAdd(remoteConfig, args[0], args[1], global); err != nil {
+		name := args[0]
+		uri := args[1]
+		if err := singularity.RemoteAdd(remoteConfig, name, uri, global); err != nil {
 			sylog.Fatalf("%s", err)
 		}
+		sylog.Infof("Remote %q added.", name)
+
+		// ensure that this was not called with global flag, otherwise this will store the token in the
+		// world readable config
+		if !global && !remoteNoLogin {
```

Comment:
Even if the flag is not changed, I would change the name of the variable: `!remoteNoLogin` means "perform login". This would be easier to read as:

```
if remoteDoLogin && !global {
```

and when written in this form I have to wonder what the case `remoteDoLogin=true, global=true` does (since it's not handled here).

Reply:
I think changing the the variable name will need a change in the flag name or an extra variable just to imply the opposite behavior. Any suggestions on this? I have also improved the logic to handle the case you presented explicitly instead of silently like before.